web/yo/app/scripts/controllers/gene.js

```diff
@@ -3009,6 +2987,119 @@ angular.module('oncokbApp')
             getOncoTreeMainTypes();
         }]
     )
+
+    .controller('ModifyTherapyCtrl', function ($scope, $modalInstance, data, _, OncoKB, $rootScope, mainUtils, FirebaseModel, $timeout, DatabaseConnector){
+        $scope.therapy = [[]];
+        $scope.addTherapyError = false;
+        $scope.loadDrugs = function($query){
+            return DatabaseConnector.getAllDrugs()
+                .then(function(result){
+                    $scope.addTherapyError = false;
+                    var drugs = result;
+                    return drugs.filter(function (drug) {
+                        return (drug.drugName.toLowerCase().indexOf($query.toLowerCase()) != -1) || (drug.synonyms.join(',').toLowerCase().indexOf($query.toLowerCase()) != -1);
+                    })
+                })
+                .catch(
+                    function (error) {
+                        $scope.therapyErrorMessage = "Sorry, the list of drugs is not available now.";
+                        $scope.addTherapyError = true;
+                    }
+                )
+        }
+        $scope.addDruginTherapy = function(tag,index){
+            $scope.therapy[index].push(tag.drugName);
+            addTherapy(index);
+        }
+
+        $scope.removeDruginTherapy = function(tag,index){
+            var id = $scope.therapy[index].indexOf(tag.drugName);
+            if (id > -1) {
+                $scope.therapy[index].splice(id, 1);
+            }
+            addTherapy(index);
+        }
+        function addTherapy(index) {
+            if ($scope.therapy.length === index + 1 ){
+                $scope.therapy.push([]);
+            }
+            valiElement(index);
+        }
+        function valiElement(index) {
+            $scope.addTherapyError = false;
+            var therapyVali = [];
+            _.each($scope.therapy, function(element) {
+                element = element.sort();
+                element = element.join(' ').trim();
+                if (element!=='')
+                    therapyVali.push(element);
+            });
+            for(var i=0;i<therapyVali.length;i++){
```

Comment:
For line 3037 ~ 3043, I think you can simply this part of code. If you want to check if 2 arrays are exactly same, you can sort them first and then compare them directly.

Reply:
It's not for checking if 2 arrays are exactly same. It's for checking if they contain same elements.